src/main/java/muramasa/antimatter/pipe/BlockPipe.java

```diff
@@ -159,7 +169,8 @@ public void onNeighborChange(BlockState state, IWorldReader world, BlockPos pos,
         if (tile != null) {
             for (Direction side : Ref.DIRECTIONS) {
                 // Looking for the side where is a neighbor was
-                if (pos.offset(side).equals(neighbor)) {
+                // Check if the block is actually air or there was another reason for change.
+                if (pos.offset(side).equals(neighbor) && world.getBlockState(neighbor) == Blocks.AIR.getDefaultState()) {
```

Comment:
This should use state.isAir instead of comparing directly to Blocks.AIR

Reply:
It should be fixed to use isAir now. Also rebased onto new master